mucommander-core/src/main/java/com/mucommander/ui/action/impl/ShowTablePopupMenuAction.java

```diff
@@ -0,0 +1,112 @@
+/*
+ * This file is part of muCommander, http://www.mucommander.com
+ *
+ * muCommander is free software; you can redistribute it and/or modify
+ * it under the terms of the GNU General Public License as published by
+ * the Free Software Foundation; either version 3 of the License, or
+ * (at your option) any later version.
+ *
+ * muCommander is distributed in the hope that it will be useful,
+ * but WITHOUT ANY WARRANTY; without even the implied warranty of
+ * MERCHANTABILITY or FITNESS FOR A PARTICULAR PURPOSE.  See the
+ * GNU General Public License for more details.
+ *
+ * You should have received a copy of the GNU General Public License
+ * along with this program.  If not, see <http://www.gnu.org/licenses/>.
+ */
+
+package com.mucommander.ui.action.impl;
+
+import com.mucommander.commons.file.AbstractFile;
+import com.mucommander.commons.file.archive.AbstractArchiveEntryFile;
+import com.mucommander.commons.file.protocol.local.LocalFile;
+import com.mucommander.core.desktop.DesktopManager;
+import com.mucommander.text.Translator;
+import com.mucommander.ui.action.AbstractActionDescriptor;
+import com.mucommander.ui.action.ActionCategory;
+import com.mucommander.ui.action.ActionDescriptor;
+import com.mucommander.ui.action.ActionProperties;
+import com.mucommander.ui.dialog.InformationDialog;
+import com.mucommander.ui.main.MainFrame;
+import com.mucommander.ui.main.menu.TablePopupMenu;
+import com.mucommander.ui.main.table.FileTable;
+import com.mucommander.ui.main.table.FileTableModel;
+
+import java.awt.Rectangle;
+import java.awt.event.KeyEvent;
+import java.util.Map;
+
+import javax.swing.KeyStroke;
+
+/**
+ * This action shows TablePopupMenu (file context menu) via keyboard shortcut (ctrl-space on file or folder).
+ * This contextual menu appears also when right-clicked on file/folder.
+ */
+public class ShowTablePopupMenuAction extends ParentFolderAction {

```

Comment:
what's the difference between this action and `ShowFilePopupMenuAction`?

Reply:
I will take a look at this one, haven't spotted it before.

Reply:
So, here are differences.

`ShowFilePopupMenuAction`:
- doesn't open contextual menu on '..'
- it places popup menu below the selected file
- it uses ActionCategory.FILES category

My version ie `ShowTablePopupMenuAction`:
- opens contextual items also on '..' - just like it would if right-clicked
- it places popup menu to the top-right of name column - imho, more userfriendly, it doesn't obstruct the view of the file (if the name is close to the bottom of the window).
- it uses ActionCategory.NAVIGATION category

My proposition is to apply my changes over ShowFilePopupMenuAction, but:
- keep `ShowFilePopupMenuAction` ACTION_ID 
- add `KeyEvent.VK_CONTEXT_MENU` to getDefaultAltKeyStroke (for Microsoft Windows Context Menu key)

What do you think?

Reply:
sounds like a plan 👍

Reply:
I've found one problem with my solution - it doesn't wok with search results (popup doesn't show) :/
I will try to figure out how have it working for both scenarios: parent folder and search results.

Reply:
PR updated. It should work on files/folders and both parent folder and search results.